docs/source/dev/tests.md

```diff
@@ -0,0 +1,94 @@
+# Automated Testing
+
+Testing is a crucial aspect of software development that ensures the reliability, quality, and performance of an application. Additionally, testing facilitates better code design and architecture by encouraging modular and decoupled code. Hence, this document will focus on the different testing frameworks used in this project, explain how to correctly use them and how to contribute to the testing environment.
+
+The tests in this project include two different frameworks, namely `pytest` and `cypress`. The focus of both testing frameworks is slightly different:
+ - `pytest` tests focus on individual units of code, for this application, these tests primarly cover authentication and routing. 
+ - `cypress` is used for e2e tests. These tests should cover the application's workflow from the user's perspective. They simulate user interaction with the application's interface and ensure the correct functionality of features, integration of components, and overall user experience.
+
+## Prerequirements
+
+Before you can run any tests, you need to make sure to run the local _Redis_ server. You can simply do so by running the `redis-server` command in the `mxcubeweb` conda environment (Best opening a new terminal for this).
+
+For running the e2e tests you need to additionally run the front-end part of MxCube, to do so please follow [these instructions](./environment.md).
+
+It is best to run all the tests in the `mxcubeweb` conda environment.
```

Comment:
Technically speaking no one needs a conda environment. I would rather leave mentions of "conda" out of this document. And instead write something along the lines of:

```suggestion
Running tests requires the Redis server to be running.

The document [](./environment.md) has more information about setting up a development environment for running tests.
```